modules/bot.py

```diff
@@ -983,42 +1034,44 @@ def pick_player(self, member, args):
             return
 
         if len(args):
-            targetid = args[0].lstrip("<@!").rstrip(">")
-            if not targetid.isdigit():
+            target_id = args[0].lstrip("<@!").rstrip(">")
+            if not target_id.isdigit():
                 client.reply(self.channel, member, "You must specify a highlight!")
                 return
-            targetid = int(targetid)
-            for i in match.unpicked:
-                if i.id == targetid:
-                    team.append(i)
-                    match.unpicked.remove(i)
-                    if len(match.unpicked) == 0 or match.maxplayers == len(match.alpha_team) + len(match.beta_team):
-                        match.next_state()
-                    elif len(match.unpicked) == 1 and match.pick_order:
+            target_id = int(target_id)
+            if target_id <= 10:
+                player = match.unpicked_pool.pick_by_position(target_id)
+            else:
+                player = match.unpicked_pool.pick_by_player_id(target_id)
+
+            if player:
+                team.append(player)
+                if len(match.unpicked_pool) == 0 or match.maxplayers == len(match.alpha_team) + len(match.beta_team):
+                    match.next_state()
+                elif len(match.unpicked_pool) == 1 and match.pick_order:
+                    match.pick_step += 1
+                    if match.pick_order[match.pick_step] == 'a':
+                        match.alpha_team.append(match.unpicked_pool.first())
+                    else:
+                        match.beta_team.append(match.unpicked_pool.first())
```

Comment:
The nice thing about this unpicked_object is that any calls that retrieve a player also remove them from the pool so a second .remove() call is not necessary like it previously did.